FWCore/Framework/interface/Principal.h

```diff
@@ -126,11 +129,17 @@ namespace edm {
                            SharedResourcesAcquirer* sra,
                            ModuleCallingContext const* mcc) const;
 
-    void prefetchAsync(WaitingTask* waitTask,
+    template <typename INFOTYPE>
+    void prefetchAsync(WaitingTask* task,
                        ProductResolverIndex index,
                        bool skipCurrentProcess,
                        ServiceToken const& token,
-                       ModuleCallingContext const* mcc) const;
+                       ModuleCallingContext const* mcc,
+                       INFOTYPE const& info) const {
+      auto const& productResolver = productResolvers_.at(index);
```

Comment:
Please do not use `at` as this is performance critical code.

Reply:
It's already using `at()`, just in the `.cc`
https://github.com/cms-sw/cmssw/blob/53c50cdd24c436cca4892e8fb7e0d5b5b59b641e/FWCore/Framework/src/Principal.cc#L631